### PR TITLE
Fixes a crash in image_save, changes cli working dir behavior to matc…

### PR DESCRIPTION
**Pull request by RemoveRusky**
…h lgm's, fixes a bug in sprite_copy, comments out some unimplented sprite funcs and adds tests for sprite resources

### CODE REVIEW

File: ENIGMAsystem/SHELL/Universal_System/image_formats.cpp [hunk 1]
```diff
@@ -320,7 +320,7 @@ int image_save_png(string filename, const unsigned char* data, unsigned width, u
         file.close();
     }
     free(buffer);
-	free(bitmap);
+    delete[] bitmap;
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Universal_System/image_formats.cpp]: Good job, `new` should always go with `delete` and `free()` only goes with `malloc()`. This file does not have any malloc calls, so there should be no calls to free:
https://stackoverflow.com/a/4061522

**Comment by RemoveRusky** [comment]
@RobertBColton is idiot and messed up this pr. Some of the shit he changed to delete should actually be free() as lodepng uses malloc but he didn't bother to look like a twat

**Comment by RobertBColton** [comment]
Yeah it looks like I made a mistake there. However, in their official examples:
http://lodev.org/lodepng/

They don't call free or delete because they use an std::vector of char. @JoshDreamland should we do that?

**Comment by JoshDreamland** [comment]
If they are using a vector in their official examples, then by all means, use a vector. The API call you're using right now does not allow for a vector. Please do not attempt to pass a vector's `data()` pointer to it.

**Pull request merged**